PostHog/PostHogStorage.swift

```diff
@@ -15,16 +15,54 @@ import Foundation
  */
 func applicationSupportDirectoryURL() -> URL {
     let url = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask).first!
+
+    #if canImport(XCTest) // only visible to test targets
+        let bundleIdentifier = Bundle.main.bundleIdentifier ?? "com.posthog.test"
+    #else
+        let bundleIdentifier = Bundle.main.bundleIdentifier!
+    #endif
+
+    return url.appendingPathComponent(bundleIdentifier)
+}
+
+/**
+
+ From Apple Docs:
+ In iOS, the value is nil when the group identifier is invalid. In macOS, a URL of the expected form is always
+ returned, even if the app group is invalid, so be sure to test that you can access the underlying directory
+ before attempting to use it.
+
+ MacOS: The system also creates the Library/Application Support, Library/Caches, and Library/Preferences
+ subdirectories inside the group directory the first time you use it
+ iOS: The system creates only the Library/Caches subdirectory automatically
+
+  see: https://developer.apple.com/documentation/foundation/filemanager/1412643-containerurl/
+  */
+func appGroupContainerUrl(config: PostHogConfig) -> URL? {
+    guard let appGroupIdentifier = config.appGroupIdentifier else { return nil }
+
     #if canImport(XCTest) // only visible to test targets
-        return url.appendingPathComponent(Bundle.main.bundleIdentifier ?? "com.posthog.test")
+        let bundleIdentifier = "com.posthog.test"
     #else
-        return url.appendingPathComponent(Bundle.main.bundleIdentifier!)
+        let bundleIdentifier = Bundle.main.bundleIdentifier!
```

Comment:
can we create a func to return the `Bundle.main.bundleIdentifier`? since this code now is dupe here and above